subprojects/gradle/upload-cd-build-result/src/main/kotlin/com/avito/cd/ProjectExtentions.kt

```diff
@@ -23,7 +24,7 @@ private class CdBuildConfigFactory : (Project) -> Provider<CdBuildConfig> {
                 Providers.of(configFilePath).map { path ->
                     val configFile = project.rootProject.file(path)
                     val config = deserializeToCdBuildConfig(configFile)
-                    validate(config)
+                    CdBuildConfigValidator(config, project.ciLogger).validate()
```

Comment:
We should configure Validator to fail a build if we have unexpected deployment but we don't wait for it